services/stats/stats.go

```diff
@@ -93,29 +97,66 @@ type RudderStatsT struct {
 	dontProcess bool
 }
 
+func getNewStatsdClientWithExpoBackoff(opts ...statsd.Option) {
```

Comment:
as mentioned in another comment, this method should return an `error` if the maximum wait time is reached.

Reply:
Will it ever reach maximum, or retry constantly at a lesser pace?

Reply:
not needed if we retry forever

Reply:
made change to return error from this function, but simply logging it in `Setup` func. Currently, since none of the `Setup's` called from main.go returns `error`. I feel it might be better to take it up together for all of them, and also handle other panics. @atzoum  WDYT?

Reply:
> made changes to return error from this function, but simply logging it in `Setup` func. Currently, since none of the `Setup's` called from main.go returns `error`. I feel it might be better to take it up together for all of them, and also handle other panics. @atzoum WDYT?

What you describe sounds good, however I cannot see it materialized in this PR

Reply:
can you check now?